src/Analyzers/AnalyzerOptionExtensions.cs

```diff
@@ -44,14 +42,17 @@ private static string GetCategoryBasedDotnetAnalyzerDiagnosticSeverityKey(string
                 return false;
             }
 
-            // If user has explicitly configured severity for this diagnostic ID, that should be respected and
-            // bulk configuration should not be applied.
+            // If user has explicitly configured severity for this diagnostic ID, that should be respected.
+            if (compilation.Options.SpecificDiagnosticOptions.TryGetValue(descriptor.Id, out severity))
+            {
+                return true;
+            }
+
+            // If user has explicitly configured severity for this diagnostic ID, that should be respected.
             // For example, 'dotnet_diagnostic.CA1000.severity = error'
-            if (compilation.Options.SpecificDiagnosticOptions.ContainsKey(descriptor.Id) ||
-                tree.DiagnosticOptions.ContainsKey(descriptor.Id))
+            if (tree.DiagnosticOptions.TryGetValue(descriptor.Id, out severity))
```

Comment:
We were not returning the severity from DiagnosticOptions causing some analyzers to not be run.

Reply:
Well, this is why we need a test for this :(